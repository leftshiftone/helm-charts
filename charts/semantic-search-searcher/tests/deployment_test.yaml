suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: default
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: metadata.name
          value: RELEASE-NAME-semantic-search-searcher-app
      - matchSnapshot:
          path: spec
  - it: custom values
    set:
      image.tag: latest
      resources.limits.cpu: 1000m
      resources.limits.memory: 1Gi
    asserts:
      - matchSnapshot:
          path: spec
