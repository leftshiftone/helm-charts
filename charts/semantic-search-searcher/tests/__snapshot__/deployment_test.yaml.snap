custom values:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: semantic-search-searcher
    strategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        annotations:
          checksum/config: 89821d3599327337cfe017747b12ab7cea6a816756b83bbbb6e79353df6a5727
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: semantic-search-searcher
          application: semantic-search-searcher
      spec:
        containers:
        - envFrom:
          - configMapRef:
              name: RELEASE-NAME-semantic-search-searcher
          image: 007098893018.dkr.ecr.eu-central-1.amazonaws.com/semantic-search-searcher:latest
          imagePullPolicy: IfNotPresent
          livenessProbe:
            httpGet:
              path: /health
              port: http
          name: semantic-search-searcher
          ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /ready
              port: http
          resources:
            limits:
              cpu: 1000m
              memory: 1Gi
          securityContext: {}
          startupProbe:
            failureThreshold: 30
            httpGet:
              path: /ready
              port: http
            periodSeconds: 10
        initContainers:
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null $ELASTICSEARCH_URL/_cluster/health;
            do echo "Waiting for elasticsearch to be available at $ELASTICSEARCH_URL/_cluster/health...";
            sleep 3; done
          env:
          - name: ELASTICSEARCH_URL
            valueFrom:
              configMapKeyRef:
                key: ES_URL
                name: RELEASE-NAME-semantic-search-searcher
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-searcher-elasticsarch-check
        securityContext: {}
        serviceAccountName: RELEASE-NAME-semantic-search-searcher
default:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: semantic-search-searcher
    strategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 0
      type: RollingUpdate
    template:
      metadata:
        annotations:
          checksum/config: 89821d3599327337cfe017747b12ab7cea6a816756b83bbbb6e79353df6a5727
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: semantic-search-searcher
          application: semantic-search-searcher
      spec:
        containers:
        - envFrom:
          - configMapRef:
              name: RELEASE-NAME-semantic-search-searcher
          image: 007098893018.dkr.ecr.eu-central-1.amazonaws.com/semantic-search-searcher:0.0.0
          imagePullPolicy: IfNotPresent
          livenessProbe:
            httpGet:
              path: /health
              port: http
          name: semantic-search-searcher
          ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /ready
              port: http
          resources: {}
          securityContext: {}
          startupProbe:
            failureThreshold: 30
            httpGet:
              path: /ready
              port: http
            periodSeconds: 10
        initContainers:
        - command:
          - sh
          - -c
          - until curl --fail --silent --output /dev/null $ELASTICSEARCH_URL/_cluster/health;
            do echo "Waiting for elasticsearch to be available at $ELASTICSEARCH_URL/_cluster/health...";
            sleep 3; done
          env:
          - name: ELASTICSEARCH_URL
            valueFrom:
              configMapKeyRef:
                key: ES_URL
                name: RELEASE-NAME-semantic-search-searcher
          image: curlimages/curl:7.84.0
          imagePullPolicy: IfNotPresent
          name: semantic-search-searcher-elasticsarch-check
        securityContext: {}
        serviceAccountName: RELEASE-NAME-semantic-search-searcher
